aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -50,6 +50,15 @@ void window_function_checks(
       window_length);
 }
 
+static inline bool allIntegral(std::initializer_list<std::reference_wrapper<Scalar>> l) {
```

Comment:
you should probably comment on the boolean case.  I think this implementation is correct (and we never use bool, which matches NumPy), but I'm not 100% sure.